### PR TITLE
fnlfmt: 0.3.1 -> 0.3.2

### PR DESCRIPTION
**Pull request by abhisheksingh0x558**
[CHANGELOG](https://git.sr.ht/~technomancy/fnlfmt/tree/main/item/changelog.md#032--2024-09-02)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by abhisheksingh0x558** [comment]
@khaneliman can you please have a look at this as well?

**Comment by abhisheksingh0x558** [comment]
@GaetanLepage can you please have a look at this?

File: pkgs/development/tools/fnlfmt/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
   pname = "fnlfmt";
-  version = "0.3.1";
+  version = "0.3.2";
 
   src = fetchFromSourcehut {
     owner = "~technomancy";
     repo = pname;
     rev = version;
-    sha256 = "sha256-rhbYG0TpqAZnbLaZOG739/pDA61Dwb4Y1HhBxWLtOus=";
+    sha256 = "sha256-wbeWAv4xhxh7M6tRd9qpgBRtg1/fqg0AUPvh2M5f60Q=";
```
Comment (khaneliman) [inline pkgs/development/tools/fnlfmt/default.nix:11]:
```suggestion
    hash= "sha256-wbeWAv4xhxh7M6tRd9qpgBRtg1/fqg0AUPvh2M5f60Q=";
```

File: pkgs/development/tools/fnlfmt/default.nix [hunk 2]
```diff
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   pname = "fnlfmt";
-  version = "0.3.1";
+  version = "0.3.2";
 
   src = fetchFromSourcehut {
     owner = "~technomancy";
     repo = pname;
```
Comment (khaneliman) [inline pkgs/development/tools/fnlfmt/default.nix:9]:
```suggestion
    repo = “fnlfmt”;
```

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349207`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

File: pkgs/development/tools/fnlfmt/default.nix [hunk 3]
```diff
@@ -2,13 +2,13 @@
...
+  version = "0.3.2";
 
   src = fetchFromSourcehut {
     owner = "~technomancy";
-    repo = pname;
-    rev = version;
-    sha256 = "sha256-rhbYG0TpqAZnbLaZOG739/pDA61Dwb4Y1HhBxWLtOus=";
+    repo = "fnlfmt";
+    rev = "refs/tags/${version}";
+    hash = "sha256-wbeWAv4xhxh7M6tRd9qpgBRtg1/fqg0AUPvh2M5f60Q=";
```
Comment (GaetanLepage) [inline pkgs/development/tools/fnlfmt/default.nix:11]:
```suggestion
    hash = "sha256-LYHhKC8iA4N8DdCH8GfSOkN/e+W3YjkFhVSDQraKoFk=";
```

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349207`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Fails with:
```
┃        > Running phase: buildPhase
┃        > /nix/store/jsnhczy4jwn4d3pkimy95kk3m8alyy4w-lua-5.2.4/bin/lua: ...l-1.5.1-1/fennel-1.5.1-1-rocks/fennel/1.5.1-1/bin/fennel:7172: cli.fnl: No such file or directory
┃        > stack traceback:
┃        >      [C]: in function 'assert'
┃        >      ...l-1.5.1-1/fennel-1.5.1-1-rocks/fennel/1.5.1-1/bin/fennel:7172: in function <...l-1.5.1-1/fennel-1.5.1-1-rocks/fennel/1.5.1-1/bin/fennel:7165>
┃        >      (...tail calls...)
┃        >      [C]: in ?
```

**Comment by abhisheksingh0x558** [comment]
> Fails with:
> 
> ```
> ┃        > Running phase: buildPhase
> ┃        > /nix/store/jsnhczy4jwn4d3pkimy95kk3m8alyy4w-lua-5.2.4/bin/lua: ...l-1.5.1-1/fennel-1.5.1-1-rocks/fennel/1.5.1-1/bin/fennel:7172: cli.fnl: No such file or directory
> ┃        > stack traceback:
> ┃        >      [C]: in function 'assert'
> ┃        >      ...l-1.5.1-1/fennel-1.5.1-1-rocks/fennel/1.5.1-1/bin/fennel:7172: in function <...l-1.5.1-1/fennel-1.5.1-1-rocks/fennel/1.5.1-1/bin/fennel:7165>
> ┃        >      (...tail calls...)
> ┃        >      [C]: in ?
> ```

Checking

**Comment by abhisheksingh0x558** [comment]
Updated the hash. `home-manager switch` with this one works fine

**Comment by abhisheksingh0x558** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349207`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349207`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
> `nixpkgs-review` result

Same error as before:
```
trying https://git.sr.ht/~technomancy/fnlfmt/archive/refs/tags/0.3.2.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  127k    0  127k    0     0   519k      0 --:--:-- --:--:-- --:--:--  520k
unpacking source archive /private/tmp/nix-build-source.drv-0/0.3.2.tar.gz
```

**Comment by abhisheksingh0x558** [comment]
> > `nixpkgs-review` result
> 
> Same error as before:
> 
> ```
> trying https://git.sr.ht/~technomancy/fnlfmt/archive/refs/tags/0.3.2.tar.gz
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 100  127k    0  127k    0     0   519k      0 --:--:-- --:--:-- --:--:--  520k
> unpacking source archive /private/tmp/nix-build-source.drv-0/0.3.2.tar.gz
> ```

It's strange that it [built](https://github.com/NixOS/nixpkgs/pull/349207#issuecomment-2425167459) for me

**Comment by GaetanLepage** [comment]
> It's strange that it [built](https://github.com/NixOS/nixpkgs/pull/349207#issuecomment-2425167459) for me

You probably have it cached locally.
Try emptying the hash and see what it gives.

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349207`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
Yeah, get the same hash mismatch.

```bash
specified: sha256-wbeWAv4xhxh7M6tRd9qpgBRtg1/fqg0AUPvh2M5f60Q=
      got: sha256-LYHhKC8iA4N8DdCH8GfSOkN/e+W3YjkFhVSDQraKoFk=

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349207`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

**Comment by abhisheksingh0x558** [comment]
@GaetanLepage are you getting this error?
┃        > Running phase: buildPhase
┃        > /nix/store/jsnhczy4jwn4d3pkimy95kk3m8alyy4w-lua-5.2.4/bin/lua: ...l-1.5.1-1/fennel-1.5.1-1-rocks/fennel/1.5.1-1/bin/fennel:7172: cli.fnl: No such file or directory
┃        > stack traceback:
┃        >      [C]: in function 'assert'
┃        >      ...l-1.5.1-1/fennel-1.5.1-1-rocks/fennel/1.5.1-1/bin/fennel:7172: in function <...l-1.5.1-1/fennel-1.5.1-1-rocks/fennel/1.5.1-1/bin/fennel:7165>
┃        >      (...tail calls...)
┃        >      [C]: in ?

**Comment by abhisheksingh0x558** [comment]
@GaetanLepage I think this [commit](https://github.com/NixOS/nixpkgs/pull/349207/commits/07bdd84f7ecf7caded58d21cd4ac040b8abd48d1) should fix it. The issue was due this [change](https://github.com/NixOS/nixpkgs/pull/349207/commits/07bdd84f7ecf7caded58d21cd4ac040b8abd48d1)

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349207`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fnlfmt</li>
  </ul>
</details>

**Comment by kirillrdy** [comment]
please squash commits

**Comment by abhisheksingh0x558** [comment]
> please squash commits

Done

**Review by GaetanLepage — APPROVED** [review]
LGTM !
Let's wait for ofborg.

**Pull request merged**